app/scheduler.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from apscheduler.schedulers.background import BackgroundScheduler
+from apscheduler.schedulers import SchedulerAlreadyRunningError, SchedulerNotRunningError
+from threading import Thread, Lock
+from multiprocessing.dummy import Pool as ThreadPool 
+from nodeodm.models import ProcessingNode
+from app.models import Task
+from django.db.models import Q
+import random
+
+logger = logging.getLogger('app.logger')
+scheduler = None
+
+def job(func):
+    """
+    Adds background={True|False} param to any function
+    so that we can call update_nodes_info(background=True) from the outside
+    """
+    def wrapper(*args,**kwargs):
+        if (kwargs.get('background', False)):
```

Comment:
So currently if you call:

```python
myfunc(background=True) # runs in background (not on main thread)
myfunc(background=False) # on main thread
myfunc() # on main thread
```

The second parameter to .get is the default value if 'background' is not provided in the dict. Perhaps I should add a comment, it's kind of difficult to read.